### PR TITLE
ref bug #130999 Error data,  bug #130997 Clear datetime

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Sửa lỗi sai dữ liệu khi click clear button on dialog
Redmine: https://dev.framgia.com/redmine/issues/130999
Sửa lỗi xóa datime khi click clear button on dialog
https://dev.framgia.com/redmine/issues/131997

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 1]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
             if (mStartDateNoSalary != null && mEndDateNoSalary == null) {
-                showErrorDialog(mContext.getString(R.string.the_field_required_can_not_be_blank));
+                //showErrorDialog(mContext.getString(R.string.the_field_required_can_not_be_blank));
+                showErrorDialog("End date not empty");
                 return;
             }
             mNavigator.startActivityForResult(ConfirmRequestOffActivity.class, bundle,
                     Constant.RequestCode.REQUEST_OFF);
         } else {
+                            if(StringUtils.isBlank(mStartDateNoSalary) && StringUtils.isBlank(
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: 'if' have incorrect indentation level 28, expected level should be 12.

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: 'if' is not followed by whitespace.

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 2]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
+                                    mEndDateNoSalary)){
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: '{' is not preceded with whitespace.

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 3]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
+                                showErrorDialog(mContext.getString(R.string.the_field_required_can_not_be_blank));
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: 'if' child have incorrect indentation level 32, expected level should be 16.

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: Line is longer than 100 characters (found 114).

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 4]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
+                            }
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: 'if rcurly' have incorrect indentation level 28, expected level should be 12.

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 5]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
             if (StringUtils.isBlank(mEndDateHaveSalary) && StringUtils.isBlank(
                     mStartDateHaveSalary)) {
+//                if(StringUtils.isBlank(mStartDateNoSalary) && StringUtils.isBlank(
+//                        mEndDateNoSalary)){
+//                    showErrorDialog(mContext.getString(R.string.the_field_required_can_not_be_blank));
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: Line is longer than 100 characters (found 104).

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 6]
```diff
@@ -1402,28 +1405,39 @@ public void onClickNext(View view) {
...
+//                    showErrorDialog(
+//                            mContext.getString(R.string.the_field_required_can_not_be_blank));
+                    showErrorDialog("Date no salary not empty");
                     return;
                 } else {
-                    if (StringUtils.isNotBlank(mEndDateHaveSalary) || StringUtils.isNotBlank(
-                            mEndDateHaveSalary)) {
-                        showErrorDialog(
-                                mContext.getString(R.string.the_field_required_can_not_be_blank));
+                    if (StringUtils.isNotBlank(mEndDateHaveSalary) ) {
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: ERROR: ')' is preceded with whitespace.

File: app/src/main/java/com/framgia/wsm/screen/managelistrequests/ManageListRequestsViewModel.java [hunk 7]
```diff
@@ -72,6 +73,8 @@
     private boolean mIsShowProgress;
     private boolean mIsVisiableLayoutDataNotFound;
     private boolean mIsVisiableLayoutFooter;
+    private boolean isVisibleFromTimeIcon;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/managelistrequests/ManageListRequestsViewModel.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isVisibleFromTimeIcon has the same name as a method
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/main/java/com/framgia/wsm/screen/managelistrequests/ManageListRequestsViewModel.java [hunk 8]
```diff
@@ -72,6 +73,8 @@
...
+    private boolean isVisibleToTimeIcon;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/managelistrequests/ManageListRequestsViewModel.java]: WARNING: [AvoidFieldNameMatchingMethodName] Field isVisibleToTimeIcon has the same name as a method
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#AvoidFieldNameMatchingMethodName

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -224,6 +224,9 @@
     <string name="end_date_must_greater_than_start_day">End day must greater than start day!</string>
     <string name="the_number_of_days_allowed">The number of days allowed for vacations may not be left blank!</string>
     <string name="the_field_required_can_not_be_blank">The field required can not be blank!</string>
+    <string name="reason_can_not_empty">Reason can not be blank !</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`reason_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -224,6 +224,9 @@
...
+    <string name="datetime_have_salary_can_not_empty">Date time have salary can not be blank !</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`datetime_have_salary_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -224,6 +224,9 @@
...
+    <string name="datetime_no_salary_can_not_empty">Date time no salary can not be blank !</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`datetime_no_salary_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java [hunk 12]
```diff
@@ -482,6 +481,9 @@ private void setCurrentBranch() {
     }
 
     private void setCurrentGroup() {
+        if (mUser.getGroups().size() == 0) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java]: pls check !=null

File: app/src/main/res/values/strings.xml (hunk 10 above)
Comment (daolq3012) [inline app/src/main/res/values/strings.xml]: pls edit to ``blank!``

File: app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -224,17 +224,20 @@
     <string name="end_date_must_greater_than_start_day">End day must greater than start day!</string>
     <string name="the_number_of_days_allowed">The number of days allowed for vacations may not be left blank!</string>
     <string name="the_field_required_can_not_be_blank">The field required can not be blank!</string>
+    <string name="reason_can_not_empty">Reason can not be blank!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`reason_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="datetime_have_salary_can_not_empty">Date time have salary can not be blank!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`datetime_have_salary_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="datetime_no_salary_can_not_empty">Date time no salary can not be blank!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`datetime_no_salary_can_not_empty`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 16]
```diff
@@ -224,17 +224,20 @@
...
-    <string name="leave_for_marriage_error">Leave for marriage is not greater than %s days !</string>
-    <string name="leave_for_child_marriage_error">Leave for child\'s marriage is not greater than %s day !</string>
-    <string name="funeral_leave_error">Funeral leave is not greater than %s days !</string>
-    <string name="leave_for_care_of_sick_error">Leave for care of sick child is not greater than %s days !</string>
-    <string name="pregnacy_examination_leave_error">Pregnancy examination leave is not greater than %s days !</string>
-    <string name="sick_leave_error">Sick leave is not greater than %s days !</string>
-    <string name="miscarriage_leave_error">Miscarriage leave is not greater than %s days !</string>
-    <string name="maternity_leave_error">Maternity leave is not greater than %s days !</string>
-    <string name="wife_labor_leave_error">Wife\'s labor leave is not greater than %s days !</string>
+    <string name="annual_leave_error">Annual leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`annual_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="leave_for_marriage_error">Leave for marriage is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`leave_for_marriage_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 18]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="leave_for_child_marriage_error">Leave for child\'s marriage is not greater than %s day!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`leave_for_child_marriage_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 19]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="funeral_leave_error">Funeral leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`funeral_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 20]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="leave_for_care_of_sick_error">Leave for care of sick child is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`leave_for_care_of_sick_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 21]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="pregnacy_examination_leave_error">Pregnancy examination leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`pregnacy_examination_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 22]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="sick_leave_error">Sick leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`sick_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 23]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="miscarriage_leave_error">Miscarriage leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`miscarriage_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 24]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="maternity_leave_error">Maternity leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`maternity_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

File: app/src/main/res/values/strings.xml [hunk 25]
```diff
@@ -224,17 +224,20 @@
...
+    <string name="wife_labor_leave_error">Wife\'s labor leave is not greater than %s days!</string>
```
Comment (trainer-android-fr) [inline app/src/main/res/values/strings.xml]: FATAL: [MissingTranslation] "`wife_labor_leave_error`" is not translated in "ja" (Japanese)
 If an application has more than one locale, then all the strings declared in one language should also be translated in all other languages.

If the string should *not* be translated, you can add the attribute `translatable="false"` on the `<string>` element, or you can define all your non-translatable strings in a resource file called `donottranslate.xml`. Or, you can ignore the issue with a `tools:ignore="MissingTranslation"` attribute.

By default this detector allows regions of a language to just provide a subset of the strings and fall back to the standard language strings. You can require all regions to provide a full translation by setting the environment variable `ANDROID_LINT_COMPLETE_REGIONS`.

You can tell lint (and other tools) which language is the default language in your `res/values/` folder by specifying `tools:locale="languageCode"` for the root `<resources>` element in your resource file. (The `tools` prefix refers to the namespace declaration `http://schemas.android.com/tools`.)

**Pull request merged**